### PR TITLE
Remove jit code of binary operation

### PR DESCRIPTION
**Pull request by karthikeyann**
- Removing jit code of binary operation which are already supported by compiled binops.
- Removed jit binops test code and benchmarks

Depends on PR https://github.com/rapidsai/cudf/pull/9766
- skips tests that require JIT code for generic PTX binary op

~disabled failing io unit tests temporarily for CI runs~ 
- cuio tests seg fault will be fixed by PR https://github.com/rapidsai/cudf/pull/9784

### CODE REVIEW

**Review by codereport — APPROVED** [review]
:fire:

File: cpp/src/binaryop/binaryop.cpp
Comment (bdice) [inline cpp/src/binaryop/binaryop.cpp:141]: There are a few lines in this file ending in empty trailing comments like `//`. It would be nice to remove those.

File: cpp/src/io/utilities/file_io_utilities.cpp
Comment (bdice) [inline cpp/src/io/utilities/file_io_utilities.cpp:79]: I approved #9784 for this fix. https://github.com/rapidsai/cudf/pull/9784#pullrequestreview-817858016

File: cpp/tests/binaryop/binop-integration-test.cpp
Comment (bdice) [inline cpp/tests/binaryop/binop-integration-test.cpp:2716]: This appears to be a very comprehensive test suite that we're deleting. Are the contents of this integration test for JIT binary operations covered by new/existing tests for compiled binary operations? (I haven't checked the scope of the tests very closely because they're 2700 lines, but I don't see an equivalent number of lines of tests for compiled binary operations unless it's just split across multiple files.)

**Review by bdice — CHANGES_REQUESTED** [review]
I have a few questions/suggestions but overall this is great. Thanks for fixing this @karthikeyann! 💯

**Comment by robertmaynard** [comment]
This needs to be marked as a breaking change as we are removing public facing API

File: cpp/tests/binaryop/binop-integration-test.cpp
Comment (karthikeyann) [reply cpp/tests/binaryop/binop-integration-test.cpp:2716]: compiled binary ops tests for most of the relevant combinations. Each tests has a comment above describing valid combinations and the types listed below in TYPED_TEST_SUITE.

Only 2 jit tests failed in compiled binary ops. (tested before removing it in this PR).
one of them is breaking change (supporting output type of boolean operation should be bool only).
another should be an unsupported type operation (bitwise operation between float and integer types).

Comment (karthikeyann) [reply cpp/tests/binaryop/binop-integration-test.cpp:2716]: Also, binops jit test code has separate tests for each type combination, while compiled binops test code is written with Typed tests. So number of lines are less because of that too.

BinaryOperationCompiledTest: 278 tests.
BinaryOperationIntegrationTest: 106 tests.

File: cpp/src/binaryop/binaryop.cpp
Comment (karthikeyann) [reply cpp/src/binaryop/binaryop.cpp:422]: updated.

File: cpp/src/binaryop/binaryop.cpp
Comment (karthikeyann) [reply cpp/src/binaryop/binaryop.cpp]: fixed. Thank you.

File: cpp/tests/binaryop/binop-integration-test.cpp
Comment (bdice) [reply cpp/tests/binaryop/binop-integration-test.cpp:2716]: Awesome, thanks for looking into that. Resolving.

**Review by bdice — APPROVED** [review]
Looks good to me.

> This needs to be marked as a breaking change as we are removing public facing API

I agree. I'll tag it as breaking.

**Review by charlesbluca — APPROVED** [review]
Approving python changes

**Review by robertmaynard — APPROVED** [review]

**Comment by ajschmidt8** [comment]
should this target `branch-21.12`? If so, let's re-target it. Otherwise the `DO NOT MERGE` label can be removed and the PR merged.

**Comment by robertmaynard** [comment]
Updating to target 21.12 as this is needed so that users ( and other PR's https://github.com/rapidsai/cudf/pull/9782 ) don't experience issues

**Comment by ajschmidt8** [comment]
superseded by #9794. closing.

**Comment by karthikeyann** [comment]
@ajschmidt8  branch-21.12 is not under active development. cudf branch-21.12 should be [frozen](https://docs.rapids.ai/maintainers). Are we still merging PRs to cudf branch-21.12 ?

**Comment by ajschmidt8** [comment]
> branch-21.12 is not under active development. cudf branch-21.12 should be [frozen](https://docs.rapids.ai/maintainers). Are we still merging PRs to cudf branch-21.12 ?

@karthikeyann, these changes were necessary for `21.12` which is why it was merged there.

**Pull request closed without merging**
